crates/ink/codegen/Cargo.toml

```diff
@@ -1,43 +1,45 @@
 [package]
 name = "ink_codegen"
-version = "4.2.0"
-authors = ["Parity Technologies <admin@parity.io>", "Robin Freyler <robin@parity.io>"]
-edition = "2021"
 
-license = "Apache-2.0"
-readme = "README.md"
-repository = "https://github.com/paritytech/ink"
+edition.workspace = true
+categories.workspace = true
+homepage.workspace = true
+keywords.workspace = true
+license.workspace = true
+repository.workspace = true
+version.workspace = true
+
+authors = ["Parity Technologies <admin@parity.io>", "Robin Freyler <robin@parity.io>"]
 documentation = "https://docs.rs/ink_codegen/"
-homepage = "https://www.parity.io/"
 description = "data structures and algorithms for generating ink! IR code"
-keywords = ["wasm", "parity", "webassembly", "blockchain", "edsl"]
-categories = ["no-std", "embedded"]
 include = ["Cargo.toml", "src/**/*.rs", "README.md", "LICENSE"]
+readme = "README.md"
 
 [lib]
 name = "ink_codegen"
 
 [dependencies]
-ink_primitives = { version = "4.2.0", path = "../../primitives" }
-ir = { version = "4.2.0", package = "ink_ir", path = "../ir", default-features = false }
-quote = "1"
-syn = { version = "2.0", features = ["parsing", "full", "extra-traits"] }
-proc-macro2 = "1.0"
-derive_more = { version = "0.99", default-features = false, features = ["from"] }
-itertools = "0.11"
-either = { version = "1.5", default-features = false }
-blake2 = "0.10"
-heck = "0.4.0"
+blake2 = { workspace = true }
+derive_more = { workspace = true, features = ["from"] }
+either = { workspace = true }
+heck = { workspace = true }
+impl-serde = { workspace = true, default-features = true }
+itertools = { workspace = true }
+# cannot inherit due to: https://github.com/bkchr/proc-macro-crate/issues/34
 scale = { package = "parity-scale-codec", version = "3.4", default-features = false, features = ["derive"] }
-impl-serde = "0.4.0"
+proc-macro2 = { workspace = true }
+quote = { workspace = true }
+serde = { workspace = true, features = ["derive"] }
+serde_json = { workspace = true }
+syn = { workspace = true, features = ["parsing", "full", "extra-traits"] }
 
-serde = { version = "1.0.137", default-features = false, features = ["derive"] }
-serde_json = "1.0.81"
+ink_ir = { workspace = true }
```

Comment:
I think returning the alias `ir` back, for now, is better to avoid conflicts. Later we can change that in a separate PR

Reply:
leaving the old version then